otherlibs/systhreads/st_stubs.c

```diff
@@ -397,12 +397,12 @@ CAMLprim value caml_thread_yield(value unit);
 
 void caml_thread_interrupt_hook(void)
 {
+  uintnat is_on = 1;
   caml_domain_state *domain = Caml_state;
   atomic_uintnat* req_external_interrupt =
     (atomic_uintnat*)&domain->requested_external_interrupt;
 
-  if (atomic_load_acq(req_external_interrupt) == 1) {
-    atomic_store_rel(req_external_interrupt, 0);
+  if (atomic_compare_exchange_strong(req_external_interrupt, &is_on, 0)) {
```

Comment:
I was expecting `atomic_exchange`. The proposed code should also work, even though it's less obvious.